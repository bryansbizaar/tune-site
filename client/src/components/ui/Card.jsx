import React from "react";

export const Card = ({ children, className, ...props }) => (
  <div className={`card ${className}`} {...props}>
    {children}
  </div>
);

export const CardHeader = ({ children, className, ...props }) => (
  <div className={`card-header ${className}`} {...props}>
    {children}
  </div>
);

export const CardTitle = ({ children, className, ...props }) => (
  <h3 className={`card-title ${className}`} {...props}>
    {children}
  </h3>
);

export const CardContent = ({ children, className, ...props }) => (
  <div className={`card-content ${className}`} {...props}>
    {children}
  </div>
);
