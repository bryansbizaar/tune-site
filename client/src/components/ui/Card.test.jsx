// import React from "react";
// import { render } from "@testing-library/react";
// import { Card } from "./Card";

describe("Card", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Card>
        <h2>Card Title</h2>
        <p>Card content</p>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
import React from "react";
import { render, screen } from "@testing-library/react";
import { Card, CardHeader, CardTitle, CardContent } from "./Card";

test("Card renders content and applies correct classes", () => {
  render(
    <Card className="test-card">
      <CardHeader className="test-header">
        <CardTitle className="test-title">Test Card</CardTitle>
      </CardHeader>
      <CardContent className="test-content">
        <p>Card content</p>
      </CardContent>
    </Card>
  );

  const card = screen.getByText("Test Card").closest(".card");
  expect(card).toHaveClass("card test-card");
  expect(screen.getByText("Test Card")).toHaveClass("card-title test-title");
  expect(screen.getByText("Card content").parentElement).toHaveClass(
    "card-content test-content"
  );
});
