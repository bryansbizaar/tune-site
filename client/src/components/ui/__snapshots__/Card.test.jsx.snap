// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card undefined"
  >
    <h2>
      Card Title
    </h2>
    <p>
      Card content
    </p>
  </div>
</DocumentFragment>
`;
